app/scripts/controllers/preferences.js

```diff
@@ -243,6 +253,13 @@ export default class PreferencesController {
     this.store.updateState({ knownMethodData });
   }
 
+  /**
+   * Clear knownMethodData from state. Used when 4Byte Resolution toggle is turned off
+   */
+  clearKnownMethodData() {
```

Comment:
I think we can consider defaulting to the contract interaction title by ignoring the cached app state instead of clearing it here.

This way, if the user reenables the toggle, they already have app state and avoid subsequent calls. You can see that'll happen by looking at these lines on `getContractMethodData` in `ui/store/actions.ts`

```js
  if (
      knownMethodData?.[fourBytePrefix] &&
      Object.keys(knownMethodData[fourBytePrefix]).length !== 0
    ) {
      return knownMethodData[fourBytePrefix];
    }
```

The user makes no privacy gains in clearing the app state here, the call has already been made.


We can implement this easily by modifying the `getKnownMethodData` in `ui/selectors/selectors.js` in this way:

```js
export function getKnownMethodData(state, data) {
  if (!data) {
    return null;
  }
  const prefixedData = addHexPrefix(data);
  const fourBytePrefix = prefixedData.slice(0, 10);
  const { knownMethodData } = state.metamask;

  const is4ByteResolutionEnabled = Boolean(state.metamask.use4ByteResolution);
  return (
    knownMethodData &&
    is4ByteResolutionEnabled &&
    knownMethodData[fourBytePrefix]
  );
}
```

Reply:
To be clear, this is just an implementation suggestion but feel free to tackle this the way it makes most sense to you, as you already have a working solution and there's no functional impact regardless of which of these two approaches you take.

Reply:
Awesome suggestion, thanks a lot! @pedronfigueiredo